### PR TITLE
Auth action to use enrolments

### PR DESCRIPTION
**Pull request by ChrisCookOC**

### CODE REVIEW

File: app/controllers/actions/IdentifierAction.scala [hunk 1]
```diff
@@ -36,39 +38,32 @@ class AuthenticatedIdentifierAction @Inject()(
...
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    authorised().retrieve(Retrievals.internalId) {
-      _.map {
-        internalId => block(IdentifierRequest(request, internalId))
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
-    } recover {
+    authorised(Enrolment(config.tgpEnrolmentIdentifier.key))
```
Comment (liandra-giacomine) [inline app/controllers/actions/IdentifierAction.scala]: can we ensure they're not agents, is it a predicate we can pass to the authorised function?
https://github.com/hmrc/trader-goods-profiles/blob/main/app/uk/gov/hmrc/tradergoodsprofiles/controllers/actions/AuthAction.scala#L73

Comment (ChrisCookOC) [reply app/controllers/actions/IdentifierAction.scala]: Added

Comment (ChrisCookOC) [reply app/controllers/actions/IdentifierAction.scala]: Done

**Review by Ewan-James-Donovan — APPROVED** [review]

**Review by yohan-oc — APPROVED** [review]
LGTM

**Review by liandra-giacomine — APPROVED** [review]

**Pull request merged**